client/ayon_shotgrid/plugins/publish/validate_shotgrid_user.py

```diff
@@ -0,0 +1,48 @@
+import pyblish.api
+from openpype.pipeline.publish import ValidateContentsOrder
+from openpype.pipeline import PublishValidationError
+
+
+class ValidateShotgridUser(pyblish.api.ContextPlugin):
+    """
+    Check if user is valid and have access to the project.
```

Comment:
Do we have plans on doing user validation at an earlier stage on the launcher and the other tools so users don't even see the projects that they don't have access instead of on the publish?

Reply:
The short answer is yes, the long one is that I believe that once the integration is more mature, we'll be able to also transfer permissions of users for projects from SG to AYON, so that way users won't be able to see the projects on the right side, currently can be achieved by removing the users from the project in Ayon manually.

Reply:
I'd say rather soon than later, but there is no specific task defined for this at the moment.